packages/stream-model-handler/src/__tests__/schema-utils.test.ts

```diff
@@ -16,12 +16,89 @@ describe('SchemaValidation', () => {
           type: 'string',
           maxLength: 80,
         },
+        objectPropName: {
+          $ref: '#/$defs/EmbeddedObject'
+        },
+      },
+      $defs: {
+        EmbeddedObject : {
+          type: 'object',
+          properties: {
+            stringPropName: {
+              type: 'string',
+              maxLength: 80
+            }
+          },
+          additionalProperties: false,
+        }
       },
-      additionallyProperties: false,
+      additionalProperties: false,
       required: ['stringPropName'],
     })).resolves.not.toThrow()
   })
-  it('returns false for an incorrect 2020-12 schema', async () => {
+
+  it('throws for correct 2020-12 schema with `additionalProperties === true` enabled on top-level', async () => {
+    expect(schemaValidator.validateSchema({
+      $schema: "https://json-schema.org/draft/2020-12/schema",
```

Comment:
it might cleanup this test file and make it a bit easier to read if you defined a valid schema object in a constant at the top of the file, then for the tests that you want to fail you can copy the valid object and then make a targeted change to the copy to make it invalid in the way you want.